libs/libcoap/Makefile

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (C) 2015 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=libcoap
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/obgm/libcoap
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)
+PKG_SOURCE_VERSION:=683337890e6fbab1eaa0ef0f2038194fb62b7fa2
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.bz2
+
+PKG_MAINTAINER:=Anton Glukhov <anton.a.glukhov@gmail.com>
+PKG_LICENSE:=GPL-2.0+
+PKG_LICENSE_FILE:=LICENSE.GPL
+
+PKG_FIXUP:=autoreconf
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/libcoap
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=CoAP(RFC 7252) library
+  URL:=http://coap.technology/
+endef
+
+define Package/libcoap/description
+  Constrained Application Protocol(RFC 7252) library
+endef
+
+TARGET_CFLAGS += $(FPIC)
+
+CONFIGURE_VARS += \
+	A2X="/usr/bin/a2x"
+
+CONFIGURE_ARGS := \
+	--host=arm-linux \
+	--target=arm-linux-gnueabi \
+	--prefix=/usr \
+	--enable-shared \
+	--enable-static \
+	--disable-examples \
+
+define Build/InstallDev
+	$(INSTALL_DIR) $(1)/usr/include
+	$(CP) $(PKG_INSTALL_DIR)/usr/include/coap/* $(1)/usr/include/
```

Comment:
Is this really what you want? So on desktop, the apps would have lines like, #include <coap/blah.h> but on OpenWrt it would be #include <blah.h> ? that doesn't seem correct....